### PR TITLE
feat: add access log for Manager API

### LINKED ISSUE #603: Do we need gin accesslog file?
image

Please take a look at this file, is it necessary in 2.0?

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

- Related issues

close #603

### CODE REVIEW

File: api/conf/conf.yaml [hunk 1]
```diff
@@ -32,6 +32,10 @@ conf:
       file_path:
         logs/error.log  # supports relative path, absolute path, standard output
                         # such as: logs/error.log, /tmp/logs/error.log, /dev/stdout, /dev/stderr
+    access_log:
+      file_path:
+        logs/access.log  # supports relative path, absolute path, standard output
+                         # such as: logs/access.log, /tmp/logs/access.log, /dev/stdout, /dev/stderr
```
Comment (membphis) [inline api/conf/conf.yaml:38]: print one example, show the format

Comment (nic-chen) [reply api/conf/conf.yaml:38]: update

**Comment by moonming** [comment]
I think you need to add unit tests.

File: api/conf/conf.go [hunk 2]
```diff
@@ -149,6 +156,17 @@ func setConf() {
 			}
 		}
 
+		// access log
+		if config.Conf.Log.AccessLog.FilePath != "" {
+			AccessLogPath = config.Conf.Log.AccessLog.FilePath
+		}
+		if !filepath.IsAbs(AccessLogPath) {
+			AccessLogPath, err = filepath.Abs(WorkDir + "/" + AccessLogPath)
```
Comment (tokers) [inline api/conf/conf.go:164]: Use `filepath.Join` instead of concatenating them manually.

File: api/filter/logging.go [hunk 3]
```diff
@@ -16,4 +16,53 @@
...
+	"go.uber.org/zap"
+)
+
+func RequestLogHandler(logger *zap.SugaredLogger) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		start, host, remoteIP, path, method := time.Now(), c.Request.Host, c.ClientIP(), c.Request.URL.Path, c.Request.Method
+		query := c.Request.URL.RawQuery
+		requestId := c.Writer.Header().Get("X-Request-Id")
+
+		blw := &bodyLogWriter{body: bytes.NewBufferString(""), ResponseWriter: c.Writer}
```
Comment (tokers) [inline api/filter/logging.go:33]: Why use `bytes.NewBufferString` but the parameter is an empty string, just use `bytes.NewBuffer` is enough.

File: api/conf/conf.go [hunk 4]
```diff
@@ -69,8 +70,14 @@ type ErrorLog struct {
 	FilePath string `yaml:"file_path"`
 }
 
+type AccessLog struct {
+	Level    string
```
Comment (tokers) [inline api/conf/conf.go:74]: log level in access log is no sense.

Comment (tokers) [reply api/conf/conf.go:74]: You can change the zapcore encoder to remove the level encoding.

**Comment by membphis** [comment]
> I think you need to add unit tests.

I think we should use E2E test here. @nic-chen please confirm

**Comment by nic-chen** [comment]
> > I think you need to add unit tests.
> 
> I think we should use E2E test here. @nic-chen please confirm


This access log is for Manager API, does not involve DP, so e2e test should not be needed.

I have added a simple unit test case to confirm that it will not cause errors.

File: api/conf/conf.go (hunk 2 above)
Comment (nic-chen) [reply api/conf/conf.go]: fixed.

File: api/conf/conf.go (hunk 4 above)
Comment (nic-chen) [reply api/conf/conf.go]: fixed.

File: api/filter/logging.go (hunk 3 above)
Comment (nic-chen) [reply api/filter/logging.go]: fixed.

File: api/filter/logging_test.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+func TestRequestLogHandler(t *testing.T) {
+        r := gin.New()
+        logger := log.GetLogger(log.AccessLog)
+        r.Use(RequestLogHandler(logger))
+        r.GET("/", func(c *gin.Context) {
+        })
+
+        w := performRequest(r, "GET", "/")
+        assert.Equal(t, 200, w.Code)
+}
```
Comment (membphis) [inline api/filter/logging_test.go:46]: please check your code editor setting, missing blank line at the end of file

**Review by membphis — COMMENTED** [review]
LGTM from the E2E test cases.

@ShiningRush @gxthrj @tokers do you have time to look at this PR ^_^

File: api/filter/logging_test.go (hunk 5 above)
Comment (nic-chen) [reply api/filter/logging_test.go]: fixed.

**Review by membphis — APPROVED** [review]
LGTM

**Review by tokers — APPROVED** [review]

**Pull request merged**